### PR TITLE
manifest: openthread upmerge up to commit a1979fd

### PR DESCRIPTION
**Pull request by maciejbaczmanski**
Cleanup of Kconfig and CMakeLists for openthread module.

Added build flags:
- `OT_15_4`
- `OT_ANYCAST_LOCATOR`
- `OT_BACKBONE_ROUTER_DUA_NDPROXYING`
- `OT_BACKBONE_ROUTER_MULTICAST_ROUTING`
- `OT_BORDER_ROUTING`
- `OT_BORDER_ROUTING_COUNTERS`
- `OT_FIREWALL`
- `OT_MESSAGE_USE_HEAP`
- `OT_NAT64_TRANSLATOR`
- `OT_TX_BEACON_PAYLOAD`

Renamed flags:
- `DATSET_UPDATER` to `DATASET_UPDATER`
- `CONFIG_OPENTHREAD_BORDER_ROUTING_NAT64` to `CONFIG_OPENTHREAD_NAT64_BORDER_ROUTING`

Rearranged flags in alphabetical order.

Signed-off-by: Maciej Baczmanski <maciej.baczmanski@nordicsemi.no>

### CODE REVIEW

**Review by canisLupus1313 — DISMISSED** [review]

**Comment by rlubos** [comment]
Is the commit mentioned in the title correct? It doesn't seem to be OT commit.

**Comment by pdgendt** [comment]
Shouldn't the new platform specific functions get an implementation?

```c
// include/openthread/platform/crypto.h
otPlatCryptoPbkdf2GenerateKey

// include/openthread/platform/radio.h
otPlatRadioAddCalibratedPower
otPlatRadioClearCalibratedPowers
otPlatRadioSetChannelTargetPower
otPlatRadioGetRawPowerSetting
```

**Comment by edmont** [comment]
> Is the commit mentioned in the title correct? It doesn't seem to be OT commit.

Right, it should be `a1979fd`.


> Shouldn't the new platform specific functions get an implementation?
> 
> ```c
> // include/openthread/platform/crypto.h
> otPlatCryptoPbkdf2GenerateKey
> 
> // include/openthread/platform/radio.h
> otPlatRadioAddCalibratedPower
> otPlatRadioClearCalibratedPowers
> otPlatRadioSetChannelTargetPower
> otPlatRadioGetRawPowerSetting
> ```

`otPlatCryptoPbkdf2GenerateKey` has a weak implementation that should work for any platform.

The power calibration feature is optional, IMO it could be implemented in a separated PR.

**Comment by maciejbaczmanski** [comment]
> > Is the commit mentioned in the title correct? It doesn't seem to be OT commit.
> 
> Right, it should be `a1979fd`.

I've copied wrong hash, sorry. Should be ok now.

**Review by edmont — DISMISSED** [review]

File: west.yml [hunk 1]
```diff
@@ -203,7 +203,7 @@ manifest:
       revision: aedcc262f93bbb1b0c2f58026911575729b7465c
       path: modules/lib/open-amp
     - name: openthread
-      revision: 3e138d3f5d549894b79cdc8567ed1f9215281344
+      revision: pull/72/head
```
Comment (rlubos) [inline west.yml:206]: Please update SHA: `a1979fdd8ed756468719cc4921b04a57efcf9f15`

**Review by edmont — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Pull request merged**